src/SearchIndex.ts

```diff
@@ -302,58 +306,31 @@ export class SearchIndex {
         assert.ok(manifest.manifestRevisionId);
 
         await session.withTransaction(async () => {
-          const operations: BulkWriteOperation<DatabaseDocument>[] = manifest.manifest.documents.map((document) => {
-            assert.strictEqual(typeof document.slug, 'string');
-            assert.ok(document.slug);
-
-            const newDocument: DatabaseDocument = {
-              ...document,
-              url: joinUrl(manifest.manifest.url, document.slug),
-              manifestRevisionId: manifest.manifestRevisionId,
-              searchProperty: [manifest.searchProperty, ...(manifest.manifest.aliases || [])],
-              includeInGlobalSearch: manifest.manifest.includeInGlobalSearch,
-            };
-
-            return {
-              updateOne: {
-                filter: { searchProperty: newDocument.searchProperty, slug: newDocument.slug },
-                update: { $set: newDocument },
-                upsert: true,
-              },
-            };
-          });
-
-          // If there are any documents in the manifest, upsert them
-          if (operations.length > 0) {
-            const bulkWriteStatus = await this.documents.bulkWrite(operations, { session, ordered: false });
-            if (bulkWriteStatus.upsertedCount) {
-              status.updated.push(manifest.searchProperty);
-            }
+          // Apply base threshold criteria for a "searchable" document and split our documents accordingly
+          const { searchable, unsearchable } = applyHeuristics(manifest.manifest.documents);
+
+          const searchableUpserts = composeUpserts(manifest, searchable);
+          const unsearchableUpserts = composeUpserts(manifest, unsearchable);
+
+          // Upsert documents deemed to be searchable, e.g. we want to surface them to users
+          if (searchableUpserts.length > 0) {
+            const bulkWriteStatus = await this.documents.bulkWrite(searchableUpserts, { session, ordered: false });
+            if (bulkWriteStatus.upsertedCount) status.updated.push(manifest.searchProperty);
+          }
+
+          // Upsert documents rejected from being searchable, for diagnostic purposes
+          if (unsearchableUpserts.length > 0) {
+            const bulkWriteStatus = await this.unindexable.bulkWrite(unsearchableUpserts, { session, ordered: false });
+            if (bulkWriteStatus.upsertedCount) status.updated.push(manifest.searchProperty);
```

Comment:
Good catch - this is potentially part of the response object at the [/status](https://docs-search-transport.docs.staging.corp.mongodb.com/status) path; no strict side effects from pushing twice, but certainly a lot of useless noise. 

I'll compose a string with more info here instead of just the searchProperty.